### DISCUSSION TITLE
Default window-padding-x should align with macOS traffic lights

### DISCUSSION THREAD

**Discussion by lelopez-io** [ghostty-org/ghostty#3174] [Feature Requests, Ideas] [answered]
### Observation
The default horizontal padding doesn't align with macOS' window traffic lights, causing terminal content to sit closer to the window edge than title bar elements. This creates a visual disconnect between the title bar and terminal content.

### Current Experience
On my MacBook Pro 14" (Nov 2024), `window-padding-x = 8` creates perfect alignment with the traffic lights. Since this scales with DPI, I'm curious if others are making similar adjustments, especially on external displays.

### Proposal
Consider adjusting the default `window-padding-x` on macOS to match native traffic light spacing. This would align with Ghostty's Zero Configuration Philosophy while maintaining platform-specific design patterns.

Would love to hear others' experiences with this spacing across different Mac setups!

<img width="912" alt="image"
src="https://github.com/user-attachments/assets/4498f4e3-1a33-4e6f-b2ce-1beb917b0485" />
*With window-padding-x = 8: Terminal content aligns with traffic lights*

<img width="912" alt="image"
src="https://github.com/user-attachments/assets/e8018abe-1eee-4deb-9f98-afa0cdea9133" />
*Default configuration: Terminal content sits closer to the window edge*

**Comment by huntercaron** [comment]
Would be nice to allow customization of this. Electron for example allows [customizating the inset](https://www.electronjs.org/docs/latest/tutorial/custom-title-bar#customize-the-traffic-light-position-macos), and you can see more insert traffic lights in apps like notes.

Ideally we could increase the padding and align the traffic lights similar to how hyper does
<img width="652" alt="Screenshot 2024-12-29 at 13 58 55" src="https://github.com/user-attachments/assets/967f63c2-dbd1-4fce-abaf-3322f36619ca" />

**Reply by lunacookies** [reply]
I think it’d be nice to use an `NSToolbar` for this purpose with config options for whether to use the standard or compact toolbar size, whether to left-align or center the window title, etc …

**Comment by 00-kat** [comment]
Defaults are to be discussed in #2363.

**Comment by narukeu** [comment]
#2363 is closed, so I can't discuss the default values in that thread anymore. I'm also in favor of setting the default `window-padding-x` to `8` on macOS for better visual consistency.

**Reply by 00-kat** [reply]
A lot has changed in the last 1.5 years; I'd normally tell you to open a new discussing instead of bumping this one but I guess this discussion works fine enough; reopening.

**Reply by narukeu** [reply, marked as answer]
> A lot has changed in the last 1.5 years; I'd normally tell you to open a new discussing instead of bumping this one but I guess this discussion works fine enough; reopening.

Opened a new thread in #13109.
